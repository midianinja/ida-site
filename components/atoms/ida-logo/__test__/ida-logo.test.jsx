import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import IDALogo from '../ida-logo';

describe('IDA Logo', () => {
  afterEach(cleanup);
  
  it('renders without crashing', async () => {
    const { getByTestId } = render(<IDALogo />);
    expect(getByTestId('ida-logo')).toHaveAttribute('src');
  });

  it('matches snapshot default button', () => { 
    const tree = renderer.create(<IDALogo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot big button', () => { 
    const tree = renderer.create(<IDALogo big />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

