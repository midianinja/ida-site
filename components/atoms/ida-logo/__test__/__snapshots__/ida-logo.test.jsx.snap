// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDA Logo matches snapshot big button 1`] = `
<img
  alt="Logo da Identidade Digital Ativista"
  className="logo __big"
  data-testid="ida-logo"
  src="/static/images/colorful-ida-logo.svg"
/>
`;

exports[`IDA Logo matches snapshot default button 1`] = `
<img
  alt="Logo da Identidade Digital Ativista"
  className="logo"
  data-testid="ida-logo"
  src="/static/images/colorful-ida-logo.svg"
/>
`;
